includes/frontend/blocks/class-crowdsignal-forms-nps-block.php

```diff
@@ -80,7 +80,13 @@ public function render( $attributes ) {
 
 		$attributes['hideBranding'] = $this->should_hide_branding();
 		$attributes['isPreview']    = is_preview();
-		$attributes['nonce']        = wp_create_nonce( self::NONCE );
+
+		// store the logged in user ID.
+		$actual_user_id = wp_get_current_user()->ID;
+		// temporarily set user ID to 0 so we create a consistent nonce.
+		wp_set_current_user( 0 );
```

Comment:
So, I still stand by my opinion this isn't really the place for this - considering our current solution works perfectly fine on WordPress.org sites.

What we should to is create a filter, say `crowdsignal_forms_api_nonce` which should default to `wp_create_nonce`.  The validation should stay the way it is.

This filter can then be used to override the nonce value when required, while keeping the optimal/default behavior for basic WP sites.